robotframework-ls/src/robotframework_ls/impl/flow_explorer_model_builder.py

```diff
@@ -74,6 +125,8 @@ def _build_hierarchy(
     suite_name: str,
     parent_body: List[Any],
     memo: dict,
+    recursion_stack: _RecursionStack,
+    user_keywords: List[Any] = None,
```

Comment:
This type isn't correct. If `user_keywords` may be `None`, it should be: `Optional[List[Any]]`.

Still, I don't follow it well, why is it `None` at all?

In the code the keyword is added to that list when it's found from a keyword call, but why doesn't it add it when it's a keyword inside a `for` or an `if`?

I'm having a bit of a hard time understanding the logic on when should a keyword be added there and when it shouldn't be added there. Can you elaborate a bit?